pages/admin/members/import.tsx

```diff
@@ -109,21 +104,53 @@ const Members = ({ user }: ServerSideProps): React.ReactElement => {
     data: tableData,
     getRowId: tableGetRowId,
   });
-
+  const [isUploading, setIsUploading] = useState(false);
+  const [isFileProcessing, setIsFileProcessing] = useState(false);
   const upload = () => {
     // remove member id to prevent data collision
-    const members = membersData?.members.map((member) => {
+    if (!membersData || membersData === undefined) {
+      toast.danger("No member data found", {
+        position: toast.POSITION.TOP_LEFT,
+      });
+      return;
+    }
+    const members = membersData.members.map((member) => {
       const result = _.omit(member, "id");
       const { memberSince } = result;
       result.memberSince = memberSince || DateTime.local().toISODate();
       return result;
     });
-    _.chunk(members, 100).forEach((people) =>
+    setIsUploading(true);
+    _.chunk(members, 100).forEach((people, batch) =>
       importPeople({
         variables: {
           people,
         },
       })
+        .then(({ data }) => {
+          data.importPeople.forEach(
+            (
+              { success, message }: { success: boolean; message: string },
+              count: number
+            ) => {
+              if (!success) {
+                toast.danger(
+                  `Error on ${members[batch * 100 + count].sid}: ${message}`,
+                  {
+                    position: toast.POSITION.TOP_LEFT,
+                  }
+                );
+              }
+            }
+          );
+          setIsUploading(false);
```

Comment:
move this to finally?

Reply:
There isn't any Promises in this part to apply `then` or `finally`

Reply:
`importPeople`?